### PR TITLE
Use defaults in jest mocks

### PR DESCRIPTION
**Pull request by fregante**
## What does this PR do?

- Followup to https://github.com/pixiebrix/pixiebrix-extension/pull/7286
- See review

## Future Work

_Done here._

## Checklist

- [x] Designate a primary reviewer: @grahamlangford

### CODE REVIEW

File: src/analysis/analysisVisitors/varAnalysis/varAnalysis.test.ts [hunk 1]
```diff
@@ -100,12 +100,7 @@ jest.mock("@/bricks/registry", () => ({
   },
 }));
 
-jest.mock("@/modDefinitions/registry", () => ({
-  __esModule: true,
-  default: {
-    lookup: jest.fn(),
-  },
-}));
```
Comment (fregante) [inline src/analysis/analysisVisitors/varAnalysis/varAnalysis.test.ts:108]: Change 2:

- Use the automated mocking (`jest.fn()` is the default)

File: eslint-local-rules/noRestrictedSyntax.test.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+const url = browser.runtime.getURL("options.html");
+
+// eslint-disable-next-line no-restricted-syntax
+const alertMock = alert as jest.Mock<typeof alert>;
+
+// eslint-disable-next-line no-restricted-syntax
+const alertMockedFunction = alert as jest.MockedFunction<typeof alert>;
+
+// eslint-disable-next-line no-restricted-syntax
+const mockPromise = jest.fn().mockResolvedValue(undefined);
```
Comment (fregante) [inline eslint-local-rules/noRestrictedSyntax.test.ts:16]: Improvement:

- Finally added tests for `no-restricted-syntax`

File: src/sidebar/sidebarSlice.test.ts [hunk 3]
```diff
@@ -30,19 +30,16 @@ import { uuidv4, validateRegistryId } from "@/types/helpers";
...
-}));
+jest.mock("@/sidebar/messenger/api");
 
 jest.mock("@/contentScript/messenger/api", () => ({
-  closeTemporaryPanel: jest.fn().mockResolvedValue(undefined),
-  cancelTemporaryPanel: jest.fn().mockResolvedValue(undefined),
-  cancelForm: jest.fn().mockResolvedValue(undefined),
+  closeTemporaryPanel: jest.fn(),
+  cancelTemporaryPanel: jest.fn(),
+  cancelForm: jest.fn(),
```
Comment (grahamlangford) [inline src/sidebar/sidebarSlice.test.ts:38]: Based on the other changes, can this not be removed as well?

Comment (fregante) [reply src/sidebar/sidebarSlice.test.ts:38]: Oh yes. I missed some because I removed jest.fn() first. Will give another look

**Review by twschiller — APPROVED** [review]

**Comment by twschiller** [comment]
@fregante LGTM! There's small conflict with https://github.com/pixiebrix/pixiebrix-extension/pull/7298 that I merged in

**Pull request merged**